[CI] CrossClusterEnrichUnavailableClustersIT testEnrichRemoteWithVendor failing

**Build Scans:**
- [elasticsearch-intake #44082 / part3](https://gradle-enterprise.elastic.co/s/z3f67is6tpurs)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:internalClusterTest" --tests "org.elasticsearch.xpack.esql.action.CrossClusterEnrichUnavailableClustersIT.testEnrichRemoteWithVendor" -Dtests.seed=6B0DC1F2A13FDCD3 -Dtests.locale=gaa-GH -Dtests.timezone=Asia/Kashgar -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.action.CrossClusterEnrichUnavailableClustersIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testEnrichRemoteWithVendor'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 1 pages have not been released
```

**Issue Reasons:**
- [main] 1 consecutive failures in test testEnrichRemoteWithVendor
- [main] 1 consecutive failures in step part3
- [main] 1 consecutive failures in pipeline elasticsearch-intake

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -547,12 +547,6 @@ tests:
 - class: org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT
   method: "test {p0=mapping/70_columnar_prefix_dynamic_properties/logsdb_columnar: multiple sub-objects with all three supported dynamic values}"
   issue: https://github.com/elastic/elasticsearch/issues/154464
-- class: org.elasticsearch.xpack.esql.action.CrossClusterEnrichUnavailableClustersIT
-  method: testEnrichRemoteWithVendor
-  issue: https://github.com/elastic/elasticsearch/issues/154532
-- class: org.elasticsearch.xpack.esql.action.CrossClusterEnrichUnavailableClustersIT
-  method: testEnrichWithHostsPolicyAndDisconnectedRemotesWithSkipUnavailableFalse
-  issue: https://github.com/elastic/elasticsearch/issues/154533
 - class: org.elasticsearch.smoketest.SmokeTestMultiNodeClientYamlTestSuiteIT
   method: "test {yaml=mapping/70_columnar_prefix_dynamic_properties/logsdb_columnar: no prefix_properties when dynamic is only set at root level}"
   issue: https://github.com/elastic/elasticsearch/issues/154537

--- a/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/datasources/datasource/TestEncryptionServicePlugin.java
+++ b/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/datasources/datasource/TestEncryptionServicePlugin.java
@@ -26,10 +26,6 @@ public class TestEncryptionServicePlugin extends Plugin {
 
     public static final String TEST_KEY_ID = "test-key";
 
-    public TestEncryptionServicePlugin() {
-        EncryptionServiceRegistry.reset();
-    }
-
     @Override
     public Collection<?> createComponents(PluginServices services) {
         EncryptionService svc = new EncryptionService() {

